### PR TITLE
Reorganize header file (cupy_cudnn.h)

### PR DESCRIPTION
**Pull request by okuta**
This PR deletes duplicate declarations.
rel #1675

### CODE REVIEW

**Comment by unnonouno** [comment]
Please merge the master

**Comment by okuta** [comment]
I merged the master.

**Comment by okuta** [comment]
I split this PR.
Please merge #1835 and #1836 before this PR.

File: cupy/cuda/cupy_cudnn.h [hunk 1]
```diff
@@ -330,26 +585,33 @@ cudnnStatus_t cudnnRNNBackwardWeights(
...
 #define cudnnConvolutionBackwardFilter_v3 cudnnConvolutionBackwardFilter
 #define cudnnSetConvolutionNdDescriptor_v3 cudnnSetConvolutionNdDescriptor
-
 #define cudnnGetFilterNdDescriptor_v5 cudnnGetFilterNdDescriptor
 
-typedef enum {} cudnnAddMode_t;
+#endif // CUDNN_VERSION >= 5000
+
+
+#if (CUDNN_VERSION >= 5000) || defined(CUPY_NO_CUDA)
```
Comment (unnonouno) [inline cupy/cuda/cupy_cudnn.h]: Why do we need to check CUPY_NO_CUDA?

**Comment by unnonouno** [comment]
You main idea is to make stub header compatible with v2 for rtd. Is it right?

File: cupy/cuda/cupy_cudnn.h (hunk 1 above)
Comment (okuta) [inline cupy/cuda/cupy_cudnn.h]: This section code are deleted over cuDNN v5.
This section code are stub for cuDNN v2.

**Comment by okuta** [comment]
Yes, it is.

File: cupy/cuda/cupy_cudnn.h (hunk 1 above)
Comment (okuta) [inline cupy/cuda/cupy_cudnn.h]: I added a comment.

File: cupy/cuda/cupy_cudnn.h [hunk 2]
```diff
@@ -6,19 +6,271 @@
...
 #ifndef CUPY_NO_CUDA
+
 #include <cudnn.h>
 
+#else // #ifndef CUPY_NO_CUDA
+
 extern "C" {
 
-#if CUDNN_VERSION < 3000
+#define CUDNN_VERSION 2000
```
Comment (gwtnb) [inline cupy/cuda/cupy_cudnn.h]: `CUDNN_VERSION` should not be defined when `CUPY_NO_CUDA` is defined.

Comment (okuta) [inline cupy/cuda/cupy_cudnn.h]: I fixed.

**Comment by okuta** [comment]
@gwtnb Please check.

**Comment by gwtnb** [comment]
LGTM

**Pull request merged**